numpyro/infer/mcmc.py

```diff
@@ -229,6 +246,8 @@ def __init__(self,
                  progress_bar=True,
                  jit_model_args=False):
         self.sampler = sampler
+        self.sample_field = sampler.sample_field
+        self.default_fields = sampler.default_fields
```

Comment:
nit: It might be better to not expose this in MCMC

Reply:
Prepended an `_` to these attributes. We are using it in a bunch of places, and `state[self.sampler.sample_field]` seems a bit much.